src/main/java/reposense/git/GitUtil.java

```diff
@@ -80,14 +86,51 @@ static String convertToGitFormatsArgs(List<Format> formats) {
     /**
      * Returns the {@code String} command to specify the globs to exclude for `git log` command.
      */
-    static String convertToGitExcludeGlobArgs(List<String> ignoreGlobList) {
+    static String convertToGitExcludeGlobArgs(File root, List<String> ignoreGlobList) {
         StringBuilder gitExcludeGlobArgsBuilder = new StringBuilder();
         final String cmdFormat = " " + addQuote(":(exclude)%s");
         ignoreGlobList.stream()
-                .filter(item -> !item.isEmpty())
+                .filter(item -> !item.isEmpty() && pathExistsInRepo(root, item))
                 .map(ignoreGlob -> String.format(cmdFormat, ignoreGlob))
                 .forEach(gitExcludeGlobArgsBuilder::append);
 
         return gitExcludeGlobArgsBuilder.toString();
     }
+
+    /**
+     * Returns true if the {@code String} path is inside the current repository
+     */
+    static boolean pathExistsInRepo(File repoRoot, String path) {
+        String validPath = path;
+        if (path.contains("/*")) {
+            validPath = path.substring(0, path.indexOf(("/*")));
+        } else if (path.contains("*")) { // not in directories
+            return true;
+        }
+        return !childIsOutsideRepo(repoRoot, new File(repoRoot, validPath));
+    }
+
+    /**
+     * Returns true if the child path is outside repository folder
+     * @throws IOException if file system queries are needed
+     */
+    static boolean childIsOutsideRepo(File repoRoot, File child) {
+        try {
+            File canonicalRoot = repoRoot.getCanonicalFile();
+            File canonicalChild = child.getCanonicalFile();
+
+            File parentFile = canonicalRoot;
+            while (parentFile != null) {
+                if (canonicalChild.equals(parentFile)) {
+                    return true;
+                }
+                parentFile = parentFile.getParentFile();
+            }
+        } catch (IOException ex) {
+            logger.log(Level.WARNING,
+                    "File/Directory in ignore glob list cannot be accessed.", ex);
```

Comment:
You are also ignoring these kinds of invalid files. So, another warning should be given when the file is ignored.

Reply:
> You are also ignoring these kinds of invalid files. So, another warning should be given when the file is ignored.

What I mean here is that you should give the user warning that there is an invalid file in ignore-glob but not just ignoring it.